src/System.Drawing.Common/src/System/Drawing/Graphics.cs

```diff
@@ -20,6 +21,369 @@ public sealed partial class Graphics : MarshalByRefObject, IDisposable, IDeviceC
+            }
+        }
+
+        public bool IsVisibleClipEmpty
+        {
+            get
+            {
+                int isEmpty;
+                int status = SafeNativeMethods.Gdip.GdipIsVisibleClipEmpty(new HandleRef(this, NativeGraphics), out isEmpty);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+
+                return isEmpty != 0;
+            }
+        }
+
+        public float PageScale
+        {
+            get
+            {
+                var scale = new float[] { 0.0f };
+                int status = SafeNativeMethods.Gdip.GdipGetPageScale(new HandleRef(this, NativeGraphics), scale);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+
+                return scale[0];
+            }
+
+            set
+            {
+                if (value <= 0 || value > 1000000032)
+                {
+                    throw new ArgumentException(SR.GdiplusInvalidParameter);
+                }
+
+                int status = SafeNativeMethods.Gdip.GdipSetPageScale(new HandleRef(this, NativeGraphics), value);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+            }
+        }
+
+        public GraphicsUnit PageUnit
+        {
+            get
+            {
+                int unit = 0;
+                int status = SafeNativeMethods.Gdip.GdipGetPageUnit(new HandleRef(this, NativeGraphics), out unit);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+
+                return (GraphicsUnit)unit;
+            }
+            set
+            {
+                if (value < GraphicsUnit.World || value > GraphicsUnit.Millimeter)
+                {
+                    throw new InvalidEnumArgumentException(nameof(value), unchecked((int)value), typeof(GraphicsUnit));
+                }
+
+                if (value == GraphicsUnit.World)
+                {
+                    throw new ArgumentException(SR.GdiplusInvalidParameter);
+                }
+
+                int status = SafeNativeMethods.Gdip.GdipSetPageUnit(new HandleRef(this, NativeGraphics), unchecked((int)value));
+                SafeNativeMethods.Gdip.CheckStatus(status);
+            }
+        }
+
+        public PixelOffsetMode PixelOffsetMode
+        {
+            get
+            {
+                PixelOffsetMode mode = 0;
+                int status = SafeNativeMethods.Gdip.GdipGetPixelOffsetMode(new HandleRef(this, NativeGraphics), out mode);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+
+                return mode;
+            }
+            set
+            {
+                if (value < PixelOffsetMode.Invalid || value > PixelOffsetMode.Half)
+                {
+                    throw new InvalidEnumArgumentException(nameof(value), unchecked((int)value), typeof(PixelOffsetMode));
+                }
+
+                if (value == PixelOffsetMode.Invalid)
+                {
+                    throw new ArgumentException(SR.GdiplusInvalidParameter);
+                }
```

Comment:
Where did this extra if block come from, and why?

Reply:
This fixes the test `PixelOffsetMode_SetToInvalid_ThrowsArgumentException` with libgdiplus